src/models.jl

```diff
@@ -496,6 +473,320 @@ metadata_model(
+const KNNFITTEDPARAMS = """
+The fields of `fitted_params(mach)` are:
+
+- `tree`: An instance of either `KDTree`, `BruteTree` or `BallTree` depending on the 
+  value of the `algorithm` hyperparameter (See hyper-parameters section above). 
+  These are data structures that stores the training data with the view of making 
+  quicker nearest neighbor searches on test data points.
+"""
+const KNNFIELDS = """
+    * `K::Int=5` : number of neighbors
+    * `algorithm::Symbol = :kdtree` : one of `(:kdtree, :brutetree, :balltree)`
+    * `metric::Metric = Euclidean()` : any `Metric` from 
+        [Distances.jl](https://github.com/JuliaStats/Distances.jl) for the 
+        distance between points. For `algorithm = :kdtree` only metrics which are 
+        instances of `$(NN.MinkowskiMetric)` are supported.
+    * `leafsize::Int = algorithm == 10` : determines the number of points 
+        at which to stop splitting the tree. This option is ignored and always taken as `0` 
+        for `algorithm = :brutetree`, since `brutetree` isn't actually a tree.
+    * `reorder::Bool = true` : if `true` then points which are close in 
+        distance are placed close in memory. In this case, a copy of the original data 
+        will be made so that the original data is left unmodified. Setting this to `true` 
+        can significantly improve performance of the specified `algorithm` 
+        (except `:brutetree`). This option is ignored and always taken as `false` for 
+        `algorithm = :brutetree`.
+    * `weights::KNNKernel=Uniform()` : kernel used in assigning weights to the 
+        k-nearest neighbors for each observation. An instance of one of the types in 
+        `list_kernels()`. User-defined weighting functions can be passed by wrapping the 
+        function in a [`UserDefinedKernel`](@ref) kernel (see ?UserDefinedKernel for more 
+        info). If observation weights `w` are passed during machine construction then the 
+        weight assigned to each neighbor vote is the product of the kernel generated 
+        weight for that neighbor and the corresponding observation weight.
+     
+    """
+
+"""
+$(MMI.doc_header(KNNClassifier))
+
+KNNClassifier implements [K-Nearest Neighbors classifier](https://en.wikipedia.org/wiki/K-nearest_neighbor_algorithm) 
+which is non-parametric algorithm that predicts a discrete class distribution associated 
+with a new point by taking a vote over the classes of the k-nearest points. Each neighbor 
+vote is assigned a weight based on proximity of the neighbor point to the test point 
+according to a specified distance metric.
+
+For more information about the weighting kernels, see the paper by Geler et.al 
+[Comparison of different weighting schemes for the kNN classifier on time-series data](https://perun.pmf.uns.ac.rs/radovanovic/publications/2016-kais-knn-weighting.pdf). 
+
+# Training data
+In MLJ or MLJBase, bind an instance `model` to data with
+    mach = machine(model, X, y)
```

Comment:
For some reason the code fencing is not kicking in here. When I view the docstring from the REPL, I'm getting

```
In MLJ or MLJBase, bind an instance model to data with mach = machine(model, X, y) OR mach = machine(model,
  X, y, w)
```

Perhaps you need spare lines around the code or you have tab character instead of 4 spaces???